### PR TITLE
Check For Executable Before Deploy & Run

### LINKED ISSUE #1671: Run instance plugin does not detect changes in Executable option
__Describe the Bug__

Run instance plugin does not detect changes made in Executable option. It allows deploying/running Non-Executable processes.

__Steps to Reproduce__

1. Try deploying/running a executable process
2. Change the process to non executable
3. Try to deploy/run again
4. It still works

__Expected Behavior__

It should throw an error

__Environment__

- Any

### PR DESCRIPTION
**Pull request by pinussilvestrus**
Closes #1671 

This checks whether an executable process is available in the diagram _before_ trying to deploy & run

**Changing Non Executable --> Executable**

Kapture 2020-01-28 at 14 16 47

**Changing Executable --> Non Executable**

Kapture 2020-01-28 at 14 17 56

### CODE REVIEW

**Comment by oguzeroglu** [comment]
Is this variable name safe to use here?
https://github.com/camunda/camunda-modeler/blob/3f225c5ac8a6f2db558aa716c7932cbb9e9e3470/client/src/plugins/camunda-plugin/start-instance-tool/StartInstanceTool.js#L143

isExecutable is also the name of the function that you import within StartInstanceTool context. Probably no big deal but just wanted to point this out 🐈

**Comment by pinussilvestrus** [comment]
I just renamed it, to avoid problems. Good catch 🎶

File: client/src/plugins/camunda-plugin/start-instance-tool/util/isExecutable.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+var moddle = new BpmnModdle();
+
+export default function isExecutable(xml) {
+
+  return new Promise(resolve => {
+    moddle.fromXML(xml, function(err, definitions) {
+
+      if (err) {
+        resolve(false);
```
Comment (barmac) [inline client/src/plugins/camunda-plugin/start-instance-tool/util/isExecutable.js]:
```suggestion
        return resolve(false);
```
We want to stop execution here.

File: client/src/plugins/camunda-plugin/start-instance-tool/__tests__/StartInstanceToolSpec.js [hunk 2]
```diff
@@ -520,7 +529,7 @@ function createStartInstanceTool({
   const wrapper = render(<TestStartInstanceTool
     subscribe={ subscribe }
     triggerAction={ triggerAction }
-    displayNotification={ noop }
+    displayNotification={ props.displayNotification || noop }
```
Comment (barmac) [inline client/src/plugins/camunda-plugin/start-instance-tool/__tests__/StartInstanceToolSpec.js]: Do we really need that if we spread the props afterwards?

File: client/src/plugins/camunda-plugin/start-instance-tool/util/isExecutable.js (hunk 1 above)
Comment (oguzeroglu) [inline client/src/plugins/camunda-plugin/start-instance-tool/util/isExecutable.js]: should we do an early return here as definitions at this point is probably undefined or null or something?

File: client/src/plugins/camunda-plugin/start-instance-tool/util/__tests__/isExecutableSpec.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+      // when
+      const executable = await isExecutable(executableXML);
+
+      // then
+      expect(executable).to.be.true;
+    });
+
+
+    it('should NOT find executable process', async function() {
```
Comment (oguzeroglu) [inline client/src/plugins/camunda-plugin/start-instance-tool/util/__tests__/isExecutableSpec.js]: Today's episode of "No big deal but": Should we rename this test to "should find/detect non-executable process" instead?

**Comment by oguzeroglu** [comment]
LGTM 👍 

If @barmac is ok too, let's merge this.

File: client/src/plugins/camunda-plugin/start-instance-tool/util/__tests__/isExecutableSpec.js (hunk 3 above)
Comment (pinussilvestrus) [reply client/src/plugins/camunda-plugin/start-instance-tool/util/__tests__/isExecutableSpec.js]: well,

What the test should do: verify no executable process is found
What the test should not do: verify a non-executable process is found

I know it's confusing, but do you know what I mean?

**Comment by oguzeroglu** [comment]
Solid job @pinussilvestrus 🔥 🔥

File: client/src/plugins/camunda-plugin/start-instance-tool/util/__tests__/isExecutableSpec.js (hunk 3 above)
Comment (oguzeroglu) [reply client/src/plugins/camunda-plugin/start-instance-tool/util/__tests__/isExecutableSpec.js]: 👍

**Pull request merged**